lib/Tag.js

```diff
@@ -1,21 +1,58 @@
 import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
 import { DragSource, DropTarget } from 'react-dnd';
 import PropTypes from 'prop-types';
 import flow from 'lodash/flow';
 
 const ItemTypes = { TAG: 'tag' };
 
 const tagSource = {
-  beginDrag: (props) => { return { id: props.tag.id } },
+  beginDrag: (props) => { return { id: props.tag.id, index: props.index } },
   canDrag: (props) => props.moveTag && !props.readOnly
 };
 
 const tagTarget = {
-  hover: (props, monitor) => {
-    var draggedId = monitor.getItem().id;
-    if (draggedId !== props.id) {
-      props.moveTag(draggedId, props.tag.id);
-    }
+  hover: (props, monitor, component) => {
+      const dragIndex = monitor.getItem().index;
+      const hoverIndex = props.index;
+
+      // Don't replace items with themselves
+      if (dragIndex === hoverIndex) {
+          return;
+      }
+
+      // Determine rectangle on screen
+      const hoverBoundingRect = findDOMNode(component).getBoundingClientRect();
+
+      // Get horizontal middle
+      const hoverMiddleX = (hoverBoundingRect.right - hoverBoundingRect.left) / 2;
+
+      // Determine mouse position
+      const clientOffset = monitor.getClientOffset();
+
+      // Get pixels to the left side
+      const hoverClientX = clientOffset.x - hoverBoundingRect.left;
+
+      // Only perform the move when the mouse has crossed half of the items width
```

Comment:
It doesn't hurt but adding comments for code that is clear by itself serves no purpose.